nodejs/lancedb/embedding/embedding_function.ts

```diff
@@ -77,6 +85,37 @@ export abstract class EmbeddingFunction<
     this.#config = {};
   }
 
+  /**
+   * Add rate limiting to the embedding function
+   * 
+   * @param maxCalls Maximum number of calls allowed in the period
+   * @param period Period in seconds
+   * @returns This embedding function for chaining
+   */
+  rateLimit(maxCalls: number = 0.9, period: number = 1.0): this {
```

Comment:
`maxCalls = 0.9` seems like a weird default if the implementation has a minimum value of 1.

Maybe you want 9 calls for every 10 seconds?